src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -54235,6 +54248,197 @@ void M(Action? a)
                 Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "a").WithLocation(10, 20));
         }
 
+        [Fact]
+        [WorkItem(44129, "https://github.com/dotnet/roslyn/issues/44129")]
+        public void DelegateCreation_FromMethodGroup_NullabilityAttributes()
```

Comment:
Consider adding some variations of this test taking into account variance. Delegate types with `out` parameters and these attributes, for example. For the same type of scenario, also consider testing usage of `MaybeNullWhen`.

Reply:
I did not test with 'out' parameters because it turns out they are not covariant in the language, but I added a test for by-value parameters and returns.

I do feel like the variance checks themselves are fairly well exercised by the overload resolution tests--delegates are just an analogous reuse of those checks, so it feels like it might be OK to not cover every last scenario